dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt

```diff
@@ -145,16 +151,17 @@ class DataManager(
         @Header(MessageHeaderKey.Type) type: String,
     ) {
         messageUtils.validateMessageType(type, MessageType.QACompleted)
-        val dataId = objectMapper.readValue(jsonString, QaCompletedMessage::class.java).identifier
+        val qaCompletedMessage = objectMapper.readValue(jsonString, QaCompletedMessage::class.java)
+        val dataId = qaCompletedMessage.identifier
         if (dataId.isEmpty()) {
             throw MessageQueueRejectException("Provided data ID is empty")
         }
         messageUtils.rejectMessageOnException {
             val metaInformation = metaDataManager.getDataMetaInformationByDataId(dataId)
-            metaInformation.qaStatus = QAStatus.Accepted
+            metaInformation.qaStatus = qaCompletedMessage.validationResult
             metaDataManager.setActiveDataset(metaInformation)
```

Comment:
An der `setActiveDataset(dataset: DataMetaInformationEntity)` würde ich zwei Sachen umbenennen:
1.) Akzeptiert sie einen param namens `dataset`, obwohl das eigentlich nur die meta infos des datasets sind, das aktiv gesetzt werden soll.  Würde ich daher umbenennen
2.) Innerhalb der Methode wird der Aktivitäts-Status des aktuell (noch) "aktiven" datasets auf "null" gesetzt. Die meta infos des aktuell (noch) "aktiven" datasets werden in eine Variable "currentlyActive" gesteckt.
Das war sehr verwirrend als ich das gelesen habe, weil der Aktivitäts-Status ebenfalls "currentlyActive" heißt, obwohl die zwei Dinge nichts miteinander zu tun haben.
Ich würde die Variable daher eher `metaInfoOfCurrentlyActiveDataset` nennen.
Dann ist das alles viel verständlicher

Reply:
Agreed. Done